src/Altinn.ApiClients.Maskinporten/Service/MaskinportenService.cs

```
@@ -0,0 +1,227 @@
+        {
+            TokenResponse token = null; ;
+
+            HttpRequestMessage requestMessage = new HttpRequestMessage()
+            {
+                Method = HttpMethod.Post,
+                RequestUri = new Uri(GetTokenEndpoint()),
+                Content = bearer
+            };
+
+            HttpResponseMessage response = await _client.SendAsync(requestMessage);
+
+            if (response.IsSuccessStatusCode)
+            {
+                string content = await response.Content.ReadAsStringAsync();
+                token = System.Text.Json.JsonSerializer.Deserialize<TokenResponse>(content);
+                return token;
+            }
+            else
+            {
+                string error = await response.Content.ReadAsStringAsync();
+            }
+
+            return null;
+        }
+
+        private string GetAssertionAud()
+        {
+            if (_maskinportenConfig.Environment.Equals("prod"))
+            {
+                return _maskinportenConfig.JwtAssertionAudienceProd;
+            }
+            else if (_maskinportenConfig.Environment.Equals("ver1"))
+            {
+                return _maskinportenConfig.JwtAssertionAudienceVer1;
+            }
+            else if (_maskinportenConfig.Environment.Equals("ver2"))
+            {
+                return _maskinportenConfig.JwtAssertionAudienceVer2;
+            }
+
+            return null;
+        }
+
+        private string GetTokenEndpoint()
+        {
+           if(_maskinportenConfig.Environment.Equals("prod"))
+            {
+                return _maskinportenConfig.TokenEndpointProd;
+            }
+            else if (_maskinportenConfig.Environment.Equals("ver1"))
+            {
+                return _maskinportenConfig.TokenEndpointVer1;
+            }
+            else if (_maskinportenConfig.Environment.Equals("ver2"))
+            {
+                return _maskinportenConfig.TokenEndpointVer2;
+            }
+
+           return null;
```

Perhaps check if a "TokenEndpoint" setting has been suppled (allowing preset values to be overridden)? If not, we should probably just throw.

---

I am not sure I understand. Since it is provided as standard it would mean they have overwritten with blank?

---

Not a big deal, but the idea was that we could make it possible for a user to supply `TokenEndpoint` and `JwtAssertionAudience` in settings as an alternative (override) to supplying `Environment`, which may be useful for testing. 

In any case, I don't think returning null here makes any sense here; it will just result in a not-immediately-obvious NullReferenceException thrown elsewhere. It should instead just `throw new ArgumentException('Invalid environment supplied, valid values are: ver1, ver2, prod')`